### ISSUE TITLE
Access error

### ISSUE THREAD

**Issue by eduard93** [intersystems-community/intersystems-servermanager#138]
I'm getting an error trying to connect to a server with a user (without %ALL):
```
This server does not have specified namespace '%SYS'.
You must select one of the following: HSCUSTOM, HSLIB, USER.
```

But how do I specify a `USER` namespace as a default?

My `settings.json` looks like this:

```json
{
    "intersystems.servers": {
        "myServer": {
            "webServer": {
                "scheme": "https",
                "host": "myaddress.com",
                "port": 443
            },
            "username": "eduard93",
        }
    }
}
``` 

In servers view I can view a list of namespaces but clicking on a `Projects` results in:

```
Cannot read properties of undefined (reading 'map')
```

**Comment by eduard93** [comment]
I have validated that these queries work under `eduard93` user (from the browser):
- /api/atelier/
- /api/atelier/v5/USER
- /api/atelier/v5/USER/docnames
- /api/atelier/v5/USER/doc/%25Api.Atelier.cls

**Comment by eduard93** [comment]
Clicking on a `Projects` as a %ALL user I see the same error. 

But in VSCode explorer everything works.

**Comment by eduard93** [comment]
Is it hard to implement this change? `USER` could be a better default.

**Comment by eduard93** [comment]
@gjsjohnmurray are there any news on this issue?

Here's a pull request with a temporary workaround, please consider merging it: https://github.com/intersystems-community/intersystems-servermanager/pull/139

**Comment by gjsjohnmurray** [comment]
@eduard93 with Global Summit over I took a closer look at this issue. My previous response misdiagnosed the cause of the message you reported.  The lines I pointed to are only used to try and pre-authenticate the Portal page when you use one of these command buttons in the Server Manager tree:

![image](https://user-images.githubusercontent.com/6726799/175542037-6e6fa3de-6dbe-4a73-a1ba-ad9619b98ef4.png)

Based on my testing with a user who doesn't have the %DB_IRISSYS:R privilege, the only consequence is that they may have to enter credentials when launching Portal from Server Manager.

Your message actually comes from the vscode-objectscript extension here:

https://github.com/intersystems-community/vscode-objectscript/blob/14ded9b136f4f643990977a15ed0781c8f1137a0/src/api/index.ts#L445-L449

I'm guessing you're using the client-side editing paradigm, not server-side (isfs), right?

What does your `objectscript.conn` settings object contain? In particular, what is its `ns` property set to?

Please check this at the User level (i.e. your local settings.json file) and at the Workspace level (i.e. .vscode/settings.json in the folder you opened in VS Code).

**Comment by eduard93** [comment]
I see in console:

```
workbench.desktop.main.js:802 TypeError: Cannot read properties of undefined (reading 'map')
	at ProjectsTreeItem.<anonymous> (c:\Users\elebedyu\.vscode\extensions\intersystems-community.servermanager-2.0.9\out\ui\serverManagerView.js:429:46)
	at Generator.next (<anonymous>)
	at fulfilled (c:\Users\elebedyu\.vscode\extensions\intersystems-community.servermanager-2.0.9\out\ui\serverManagerView.js:5:58)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by eduard93** [comment]
That's [here](https://github.com/intersystems-community/intersystems-servermanager/blob/9ac5fb002839be121d58fd45744a2bd8659fa004/src/ui/serverManagerView.ts#L496).

My user can execute `SELECT Name, Description FROM %Studio.Project`, but it returns no results.

I just need to browse the server.

**Comment by gjsjohnmurray** [comment]
@eduard93 please try the dev VSIX from the zip at https://github.com/intersystems-community/intersystems-servermanager/suites/7134425170/artifacts/283337062

With this, when you expand the Projects folder of a namespace and your user account doesn't have the %All role you should now see an error notification telling you the GRANT statement you need to run in that namespace (as a suitably privileged user).

**Comment by gjsjohnmurray** [comment]
@eduard93 the correction for this is now in 2.0.10-beta.3 which you can get at https://github.com/intersystems-community/intersystems-servermanager/releases/download/v2.0.10-beta.3/servermanager-2.0.10-beta.3.vsix

**Comment by eduard93** [comment]
Thank you @gjsjohnmurray !

**Comment by gjsjohnmurray** [comment]
@eduard93 we have now published 2.0.10
